File: 0009-palindrome-number/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
    if (x < 0 || (x % 10 === 0 && x !== 0)) {
        return false;
    }

    let realnum = x;
    let revnum = 0;

    while (x > 0) {
        let remnum = x % 10;
        revnum = revnum * 10 + remnum;
        x = Math.floor(x / 10);
    }

    if (revnum === realnum) {
        return true;
    }

    return false;
};

File: 0009-palindrome-number/0009-palindrome-number.c
#define FALSE false /*Macro defination*/
#define TRUE true

bool isPalindrome(int x) {
    if (x < 0 || (x % 10 == 0 &&
                  x != 0)) { /*Negetive number never become Palindrome number*/
        return FALSE;
    }

    int realnum = x;
    int revnum = 0;

    while (x > 0) {
        int remnum = x % 10;

        if (revnum > (INT_MAX - remnum) / 10) {
            return false; // Overflow detected
        }

        revnum = revnum * 10 + remnum;
        x /= 10;
    }

    if (realnum == revnum) {
        return TRUE;
    }

    return FALSE;
}